lisa/sut_orchestrator/azure/common.py

```diff
@@ -225,6 +354,63 @@ class AzureVmMarketplaceSchema(AzureImageSchema):
     def __hash__(self) -> int:
         return hash(f"{self.publisher}/{self.offer}/{self.sku}/{self.version}")
 
+    def _get_info(self, platform: "AzurePlatform") -> Optional[Dict[str, Any]]:
+        for location in platform._find_marketplace_image_location():
```

Comment:
If the methods `_find_marketplace_image_location` and `_get_image_info` are not called in other places in platform, they should be moved here. If there are many methods involved, we can keep as is, but remove underscore of them.

Reply:
More logic could eventually be moved to the AzureImageSchemas but I'm a bit afraid to change any of the behavior in platform because it is impossible for me to test every scenario for bugs.